designer/server/src/test/scala/pl/touk/nussknacker/test/utils/domain/TestProcessingTypeDataProviderFactory.scala

```diff
@@ -34,7 +49,44 @@ object TestProcessingTypeDataProviderFactory {
       )
     )
 
-  def fromState[T, C](stateValue: ProcessingTypeDataState[T, C]): ProcessingTypeDataProvider[T, C] =
-    new ProcessingTypeDataProvider[T, C](stateValue)(IORuntime.global) {}
+  def create(
+      processingTypeConfigs: ProcessingTypeConfigs,
+      modelClassLoaderProvider: ModelClassLoaderProvider,
+      modelDependencies: ModelDependencies,
+      deploymentManagersClassLoader: DeploymentManagersClassLoader,
+      deploymentManagerDependencies: DeploymentManagerDependencies,
+  ): ProcessingTypeDataProvider[ProcessingTypeData, CombinedProcessingTypeData] = {
+    val finalProcessingTypeData =
+      ModelDataLoader
+        .load(
+          processingTypeConfigs,
+          _ => modelDependencies,
+          modelClassLoaderProvider
+        )
+        .transform { case (modelDataWithInputs, _) =>
+          val deploymentData = DeploymentManagersLoader
```

Comment:
1. I know it's test code, but maybe we can move the `DeploymentManagersLoader.load(...)` and `ProcessingTypeDataStateFactory.create` to private methods for the sake of readability? 

2. let's avoid using _1 by doing it like that: 
```
val (deploymentData, _) = ...
```